common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -141,7 +141,7 @@ impl Navigator for NavigationController {
             annotation_json,
         };
         let next_advance = Arc::clone(&self.config.step_advance_condition);
-        NavState::new(trip_state, next_advance)
+        NavState::new(trip_state, next_advance, None)
```

Comment:
Question for this (and maybe the one above?) as they are both in `get_initial_state`:

Do we actually want/need the ability to set recording events with `NavState::new`? I guess this might be for cases where we may carry over a session across multiple controller instances (a new "trip" due to recalculating)?